relay/polkadot/src/lib.rs

```diff
@@ -1576,6 +1576,113 @@ impl pallet_rc_migrator::Config for Runtime {
 	type AhExistentialDeposit = AhExistentialDeposit;
 	type RcWeightInfo = ();
 	type AhWeightInfo = ();
+	type RcCallEnabledDuringMigration = CallsEnabledDuringMigration;
+	type RcCallEnabledAfterMigration = CallsEnabledAfterMigration;
+}
+
+/// Contains all calls that are enabled during the migration.
+pub struct CallsEnabledDuringMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledDuringMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (during, _after) = call_disable_status(call);
+		!during
+	}
+}
+
+/// Contains all calls that are enabled after the migration.
+pub struct CallsEnabledAfterMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledAfterMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (_during, after) = call_disable_status(call);
+		!after
+	}
+}
+
+/// Return whether a call should be disabled during and/or after the migration.
+///
+/// Time line looks like this:
+///
+/// --------|-----------|--------->
+///       Start        End
+///
+/// We now define 2 periods:
+///
+/// 1. During the migration: [Start, End]
+/// 2. After the migration: (End, ∞)
+///
+/// Visually:
+///
+///         |-----1-----|
+///                      |---2---->
+/// --------|-----------|--------->
+///       Start        End
+///
+/// This call returns a 2-tuple to indicate whether a call is disabled during these periods.
+pub fn call_disable_status(call: &<Runtime as frame_system::Config>::RuntimeCall) -> (bool, bool) {
+	use RuntimeCall::*;
+	const ON: bool = false;
+	const OFF: bool = true;
+
+	match call {
+		System(..) => (OFF, ON),
```

Comment:
Should we consider an escape hatch for System? If we bork the migration for some unknown reason and get stuck and don't get to the `MigrationDone` stage for more than an era (or some reasonable time period) we should allow some subset of root-only calls to rescue ourselves from doomsday gov. CC @bkontur

Reply:
I can enable it since it should not mess with the migration.